docs/components/CopyToClipboardButton.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useRef, useEffect, useState } from 'react';
+import { IconButton, Toast } from '@deque/cauldron-react';
+
+interface CopyToClipboardButtonProps
+  extends React.HTMLAttributes<HTMLButtonElement> {
+  label?: string;
+  value: string;
+}
+
+function copyTextToClipboard(text: string) {
+  const element = document.createElement('textarea');
+  element.value = text;
+  element.setAttribute('aria-hidden', 'true');
+  document.body.appendChild(element);
+
+  element.select();
+
+  let copied;
+  try {
+    copied = document.execCommand('copy');
+  } catch (ex) {
+    copied = false;
+  }
+
+  element.remove();
+
+  return copied;
+}
+
+export default function CopyToClipboardButton({
+  value,
+  label = 'copy to clipboard',
+  ...props
+}: CopyToClipboardButtonProps) {
+  const ref = useRef<HTMLButtonElement>();
+  const [accessibleName, setAccessibleName] = useState(label);
+  const [showToast, setShowToast] = useState(false);
+  const handleClick = () => {
+    copyTextToClipboard(value);
+    setShowToast(true);
+  };
+
+  const handleDismiss = () => {
+    setShowToast(false);
+    ref.current?.focus();
+  };
+
+  useEffect(() => {
+    // We don't know what context this button will be included in, so providing
+    // a minimal accessible name of "example, x of y"
+    const elements = Array.from(
+      document.querySelectorAll('[data-copy-example]')
+    );
+    const index = elements.findIndex(element => element === ref.current);
+    if (index !== -1 && elements.length) {
+      setAccessibleName(`${label} example, ${index + 1} of ${elements.length}`);
+    }
+  });
```

Comment:
There aren't any dependencies we could add here to avoid an effect on every render?  Maybe `value`?

Reply:
I realize updating every render is probably overkill, but I wanted to be sure that the _x of y_ value was always up to date with whatever other components were rendered on the page. It seems to work fine with `[value]` though 🤷